src/executables/sysrepo-plugind.c

```diff
@@ -323,13 +336,96 @@ load_plugins(struct srpd_plugin_s **plugins, int *plugin_count)
         /* finally store the plugin */
         (*plugins)[*plugin_count].handle = handle;
         (*plugins)[*plugin_count].private_data = NULL;
+
+        name_len = length_without_extension(ent->d_name);
+        if (name_len == 0) {
+            error_print(0, "Wrong filename \"%s\".", ent->d_name);
+            dlclose(handle);
+            rc = -1;
+            break;
+        }
+
+        new_str = strndup(ent->d_name, name_len);
+        if (!new_str) {
+            error_print(0, "strndup() failed.");
+            dlclose(handle);
+            rc = -1;
+            break;
+        }
+
+        (*plugins)[*plugin_count].plugin_name = new_str;
+
         ++(*plugin_count);
     }
 
     closedir(dir);
     return rc;
 }
 
+static void
+swap(struct srpd_plugin_s *a, struct srpd_plugin_s *b)
+{
+    if (a == b) {
+        return;
+    }
+    struct srpd_plugin_s tmp = *a;
+    *a = *b;
+    *b = tmp;
+}
+
+static int
+plugin_names_cmp(const struct srpd_plugin_s *plugin, const char *str2)
+{
+    /* str1 does not have the filename extension */
+    const char *str1 = plugin->plugin_name;
+    size_t n;
+
+    if (str1 == str2) {
+        return 0;
+    }
+    n = length_without_extension(str2);
+    return strlen(str1) == n ? strncmp(str1, str2, n) : -1;
```

Comment:
Based on our coding style, every boolean expression should be enclosed in parentheses.